src/jdk.compiler/share/classes/com/sun/tools/javac/tree/DocTreeMaker.java

```diff
@@ -223,9 +226,16 @@ public int getSourcePos(int index) {
                 return Position.NOPOS;
             }
 
+            // Currently, the style for a synthetic comment is not user-visible
+            // through the {@link DocTrees} API. Nevertheless, we heuristically
+            // infer a suitable style, based on the {@code preamble}, {@code postamble},
+            // and {@code fullBody}.
             @Override
             public CommentStyle getStyle() {
-                return CommentStyle.JAVADOC;
+                return preamble.isEmpty()
+                        && postamble.isEmpty()
+                        && fullBody.stream().anyMatch(t -> t.getKind() == Kind.MARKDOWN)
+                        ? CommentStyle.JAVADOC_LINE : CommentStyle.JAVADOC_BLOCK;
```

Comment:
I'm not worried about false positive `JAVADOC_LINE` than maybe false _negative_ `JAVADOC_LINE`.  Generally, the underlying issue here is how to handle weird combinations of `DocTrees` constructed by a user of the public API.  For example, should we check, and reject, a `fullBody` containing `MARKDOWN` nodes and non-empty  `preamble` or `postamble`, since that combination will never come from `DocCommentParser`.

I'm not worried about comparison with `getText` and `getSourcePos`, since there really is no other value for the source text or source position that we can return. But we can infer a best guess for the style.

Arguably, we should check the `tags` as well.

------

I dug deeper.

We do create synthetic trees in the standard doclet, such as for the descriptions of mandated methods (like `values` and `valueOf` for any enum class.  Those synthetic trees do utilize this code path, although the `getStyle` method is not currently invoked. (Verified by changing code to throw `UnsupportedOperationException` and run all the tests.) It's also true those synthetic trees do not leverage any Markdown support at this time.

I prefer to leave the code as-is at this time.

Reply:
I revisited this, and rewrote the code and added more comments for now and for later.